packages/tina-graphql/src/datasources/github-manager.ts

```diff
@@ -68,32 +68,36 @@ export class GithubManager implements DataAdaptor {
   }
   readDir = async (path: string): Promise<string[]> => {
     return _.flatten(
-      await this.cache.get(this.generateKey(path), async () =>
-        this.appOctoKit.repos
-          .getContent({
-            ...this.repoConfig,
-            path,
-          })
-          .then(async (response) => {
-            if (Array.isArray(response.data)) {
-              return await Promise.all(
-                await response.data.map(async (d) => {
-                  if (d.type === 'dir') {
-                    const nestedItems = await this.readDir(d.path)
-                    return nestedItems.map((nestedItem) => {
-                      return p.join(d.name, nestedItem)
-                    })
-                  }
-                  return d.name
-                })
-              )
-            }
+      (
+        await this.cache.get(this.generateKey(path), async () =>
+          this.appOctoKit.repos
+            .getContent({
+              ...this.repoConfig,
+              path,
+            })
+            .then(async (response) => {
+              if (Array.isArray(response.data)) {
+                return await Promise.all(
+                  await response.data.map(async (d) => {
+                    if (d.type === 'dir') {
+                      const nestedItems = await this.readDir(d.path)
+                      return nestedItems.map((nestedItem) => {
+                        return p.join(d.name, nestedItem)
+                      })
+                    }
+                    return d.name
+                  })
+                )
+              }
 
-            throw new Error(
-              `Expected to return an array from Github directory ${path}`
-            )
-          })
+              throw new Error(
+                `Expected to return an array from Github directory ${path}`
+              )
+            })
+        )
       )
+        .toString()
+        .split(',')
```

Comment:
Curious what the result was that made this necessary?

Reply:
The lru cache could store both string and arrays, so the get call would return the appropriate one. With s3 everything is a string. And the logic of requesting a file vs a directory isn't on the cache level, so we can't just split in the cache. If we wanted to mimic the lru behaviour, maybe we format arrays we store in a way that we can then assume they are arrays by parsing them in the cache logic. This was the simplest fix, but it does feel icky and I'm not sure if it's compatible if we switched back the using the lru cache. Thoughts?

Reply:
Makes perfect sense, just wasn't sure! I don't think it matters too much to adhere to what's already there - this will be something we should in theory be able to change implementations of dramatically without it affecting anything.

Reply:
k, I am worried now that it won't work with lru so I'll give it a try to make sure

Reply:
actually never mind, .toString() will create a comma delimitated string out of an array, thus only being a pointless extra step for lru, but not a problem